pkg/avancement/service_manager.go

```diff
@@ -87,6 +90,25 @@ func (s *ServiceManager) Promote(serviceName, fromURL, toURL, newBranchName stri
 		}
 	}(keepCache, &reposToDelete)
 
+	var localSource git.Source
+	var errorSource error
+	if fromLocalRepo(fromURL) {
+		localSource = s.localFactory(fromURL, s.debug)
+	} else {
+		source, errorSource = s.checkoutSourceRepo(fromURL, fromBranch)
```

Comment:
did you mean to change 'err' to 'errorSource' here? It's only a tiny change, I just wondered if it was intentional?

Reply:
It was intentional, its to  do with `err` being assigned inside and outside of so many if statements that it causes a go panic if I leave this variable as `err`.. as in some if statements it's expecting it to be already defined and in others it isn't.

Reply:
Great, thanks :)